fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryIndexPlan.java

```diff
@@ -71,11 +71,18 @@
     protected final ScanComparisons comparisons;
     protected final boolean reverse;
 
-    public RecordQueryIndexPlan(@Nonnull final String indexName, @Nonnull IndexScanType scanType, @Nonnull final ScanComparisons comparisons, final boolean reverse) {
+    protected final boolean fullySorted;
+
+    public RecordQueryIndexPlan(@Nonnull final String indexName, @Nonnull IndexScanType scanType, @Nonnull final ScanComparisons comparisons, final boolean reverse, final boolean fullySorted) {
```

Comment:
Hmm, so it looks like that if you don't call this constructor the fully sorted property is not set. That makes problematic for the new planner. Can we either compute that property in a common code path or add some logic to `DataAccessRule`?

Reply:
Given the current structure, it seemed to make more sense to do this in `RemoveSortRule`, which is currently responsible for lining up the declared sort and the candidate scan's sort. It now additionally sets the new property when appropriate.